### DISCUSSION TITLE
Axial Induction for a given turbine v3

### DISCUSSION THREAD

**Discussion by JackDI1** [NatLabRockies/floris#937] [Q&A] [answered]
Hi all,

I am attempting to calculate the wake centerline for each turbine in a 3x3 wind farm using the Empirical Gauss deflection model (v3). The arguments of this require the thrust coefficient and the wake-induced mixing term for the ith turbine. The trust coefficient I have managed to get using `get_turbine_Cts()`, and it looks like I need to use the wake induced mixing function [here](https://github.com/NREL/floris/blob/v3/floris/simulation/wake_turbulence/wake_induced_mixing.py) to get the mixing function for the turbine, but this requires the Axial induction factor of the ith turbine as an argument. I can't seem to work out how to get this value from floris, can someone please point me in the right direction? 

If anyone has an alternative method of calculating the wake centerline I would also be interested to hear that!

Thanks,
Jack

**Comment by misi9170** [comment, marked as answer]
Hi @JackDI1 ,

To get the axial induction factors, use the [`get_turbine_ais()` method of the `FlorisInterface`](https://github.com/NREL/floris/blob/v3/floris/tools/floris_interface.py#L667). 

However, you could also use the [`sample_flow_at_points()` method of the `FlorisInterface`](https://github.com/NREL/floris/blob/v3/floris/tools/floris_interface.py#L978), and set the `x`, `y`, `z` array arguments to be locations along the wake centerline, to get the flow velocities without needing to explicitly compute the thrust coefficients and axial induction factors. See also [v3 example 28](https://github.com/NREL/floris/blob/v3/examples/28_extract_wind_speed_at_points.py) for example code.

**Reply by JackDI1** [reply]
Ah, not sure how I missed that!! Thank you!

**Comment by misi9170** [comment]
For future reference for anyone looking at this thread, in FLORIS v4, the equivalent methods are:
- [`get_turbine_thrust_coefficients()`](https://github.com/NREL/floris/blob/main/floris/floris_model.py#L935)
- [`get_turbine_ais()`](https://github.com/NREL/floris/blob/main/floris/floris_model.py#L914)
or
- [`sample_flow_at_points()`](https://github.com/NREL/floris/blob/main/floris/floris_model.py#L1282)

and an example demonstrating the use of `sample_flow_at_points()` is [examples_get_flow/002_extract_wind_speed_at_points.py](https://github.com/NREL/floris/blob/main/examples/examples_get_flow/002_extract_wind_speed_at_points.py).
